### DISCUSSION TITLE
ipopt instability, use casadi with gurobi

### DISCUSSION THREAD

**Discussion by parshakova** [casadi/casadi#3450] [General]
is it possible to use gurobi solver for nonlinear optimization in casadi?
can anyone help me with tan example of how to do so?

i am having numerical instability issues with ipopt solver

**Comment by brunoml5** [comment]
Hi! Take a look at this [page](https://github.com/casadi/casadi/wiki/FAQ:-how-to-get-third-party-solvers-to-work%3F) from the [wiki](https://github.com/casadi/casadi/wiki/Onboarding-Guide). 

Are you using python? Make sure `gurobipy` is installed in the same environment. Take a look at the installation [instructions](https://support.gurobi.com/hc/en-us/articles/360044290292-How-do-I-install-Gurobi-for-Python-#h_01GRC387FD2QYYH9J6C252RKFT) for gurobi.

**Comment by parshakova** [comment]
Thank you for your reply!

Yes, I am using Python, and have `gurobi`, and `gurobipy` installed. Also variables `GRB_LICENSE_FILE` and `DYLD_LIBRARY_PATH` are set.

However, I get the error `RuntimeError: .../casadi/core/plugin_interface.hpp:292: Plugin 'gurobi' is not found.` because
of the absence of `libcasadi_nlpsol_gurobi.dylib` file in the casadi interface library. 

Could you please let me know where I can find that file?

```python
from casadi import *

# Define symbols
x = SX.sym('x')
y = SX.sym('y')
f = x**2 + y**2  # Objective function

# Formulate optimization problem
nlp = {'x': vertcat(x, y), 'f': f, 'g': x + y - 1}
opts = {'verbose': True, 'ipopt.print_level': 0}

# Create an NLP solver object
solver = nlpsol('solver', 'gurobi', nlp, opts)
```

the error
`PluginInterface::load_plugin: Cannot load shared library 'libcasadi_nlpsol_gurobi.dylib':`

**Reply by brunoml5** [reply]
Hi!

First, gurobi is not available for NLP, only for conic solver, as can be seen [here](https://web.casadi.org/python-api/#qp). This might be one of the problems.

I tried using gurobi in my installation and I got the same error as you. I'm using Windows and conda, so it is not the same as yours.

Take a look at this [post at the forum](https://github.com/casadi/casadi/discussions/3398#discussion-5694165). The user has the same problem as you and he has a Mac. 

Gillis commented [here](https://github.com/casadi/casadi/discussions/3398#discussioncomment-7177614) that the [installation instructions](https://github.com/casadi/casadi/wiki/FAQ:-how-to-get-third-party-solvers-to-work%3F) is outdated. Hopefully he appears here and helps you.

I'm sorry that I can't help you further.
